src/client/camera.cpp

```diff
@@ -706,19 +707,20 @@ void Camera::drawNametags()
 		if (transformed_pos[3] > 0) {
 			std::wstring nametag_colorless =
 				unescape_translate(utf8_to_wide(nametag->nametag_text));
-			core::dimension2d<u32> textsize =
-				g_fontengine->getFont()->getDimension(
+			core::dimension2d<u32> textsize = font->getDimension(
 				nametag_colorless.c_str());
 			f32 zDiv = transformed_pos[3] == 0.0f ? 1.0f :
 				core::reciprocal(transformed_pos[3]);
-			v2u32 screensize = RenderingEngine::get_video_driver()->getScreenSize();
 			v2s32 screen_pos;
 			screen_pos.X = screensize.X *
 				(0.5 * transformed_pos[0] * zDiv + 0.5) - textsize.Width / 2;
 			screen_pos.Y = screensize.Y *
 				(0.5 - transformed_pos[1] * zDiv * 0.5) - textsize.Height / 2;
 			core::rect<s32> size(0, 0, textsize.Width, textsize.Height);
-			g_fontengine->getFont()->draw(
+			core::rect<s32> bg_size(-2, 0, textsize.Width+2, textsize.Height);
+			driver->draw2DRectangle(video::SColor(50, 0, 0, 0),
```

Comment:
The background should also probably change color depending on the lightness.

Maybe something like this? Note that this is psuedocode, I can't remember the exact type and member names for SColor.

Source of weights: https://stackoverflow.com/questions/3942878/how-to-decide-font-color-in-white-or-black-depending-on-background-color

```c++
core::SColor textColor = nametag->nametag_color;
bool darkBackground = (textColor.red*0.299 + textColor.green*0.587 + textColor.blue*0.114) > 186;
const core::SColor backgroundColor = darkBackground ? video::SColor(50, 0, 0, 0) : video::SColor(50, 255, 255, 255);